### ISSUE TITLE
Building a test bundle fails with 5.3 snapshots

### ISSUE THREAD

**Issue by MaxDesiatov** [swiftwasm/swift#1824]
The error is raised by `wasm-ld`:
```
wasm-ld: error: 
/Users/maxd/.carton/sdk/wasm-5.3-SNAPSHOT-2020-09-23-a/usr/share/wasi-sysroot/lib/wasm32-wasi/
libc.a(__original_main.o): undefined symbol: main
```

The build command is 
```
/Users/maxd/.carton/sdk/wasm-5.3-SNAPSHOT-2020-09-23-a/usr/bin/swift \
  build -c debug --build-tests --destination /Users/maxd/.carton/sdk/wasm-5.3-SNAPSHOT-2020-09-23-a.json -Xswiftc -color-diagnostics
```

**Comment by MaxDesiatov** [comment]
Looks we don't have any 5.3 snapshots where tests link successfully. The latest development snapshot that doesn't have this bug is `wasm-DEVELOPMENT-SNAPSHOT-2020-08-11-a`. `wasm-DEVELOPMENT-SNAPSHOT-2020-09-23-a`  fails to link with `libFoundation.a(Data.swift.obj): undefined symbol: _withStackOrHeapBuffer`.

**Comment by kateinoigakukun** [comment]
This will be fixed by cherry-picking @MaxDesiatov 's commit https://github.com/swiftwasm/swift-package-manager/commit/a36331a3dc13a9a5aba7bf91cedb7d5bb82f601e

It's just a forgetting of cherry-picking from master branch.

**Comment by MaxDesiatov** [comment]
Thanks a lot for picking this up, and sorry I forgot about that patch 😅

**Comment by MaxDesiatov** [comment]
I confirm that it was fixed in `wasm-5.3-SNAPSHOT-2020-09-25-a`.
